src/gui/SetUserStatusView.qml

```diff
@@ -0,0 +1,180 @@
+import QtQuick 2.6
+import QtQuick.Dialogs 1.3
+import QtQuick.Layouts 1.15
+import QtQuick.Controls 2.15
+import QtQuick.Window 2.15
+import com.nextcloud.desktopclient 1.0 as NC
+
+ColumnLayout {
+    id: rootLayout
+    anchors.fill: parent
+    spacing: 0
+    Layout.margins: 0
+
+    property NC.UserStatusDialogModel userStatusDialogModel
+
+    Connections {
+        target: userStatusDialogModel
+        function onShowError() {
+            errorDialog.open()
+        }
+    }
+
+    Text {
+        Layout.margins: 0
+        Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+        font.bold: true
+        text: qsTr("Online status")
+    }
+        
+    GridLayout {
+        Layout.leftMargin: 8
+        Layout.rightMargin: 8
+        Layout.topMargin: 0
+        Layout.bottomMargin: 0
+        Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+        columns: 2
+        rows: 2
+        columnSpacing: 8
+        rowSpacing: 8
+
+        Button {
+            Layout.fillWidth: true
```

Comment:
needs `implicitWidth: 100` for all buttons to have an equal size computed by the layout otherwise the button with "Do not disturb" text is expanded to a greater width than the second column